agasc/agasc.py

```diff
@@ -507,7 +541,27 @@ def get_agasc_cone(
     # Possibly expand initial radius to allow for slop due proper motion
     rad_pm = radius + (0.1 if pm_filter else 0.0)
 
-    stars = get_stars_func(ra, dec, rad_pm, agasc_file, cache)
+    # Ensure that the columns we need are read from the AGASC file, excluding PMCORR
+    # columns if supplied since they are not in the HDF5.
+    columns_query = (
+        None
+        if columns is None
+        else tuple(
+            column for column in columns if column not in ("RA_PMCORR", "DEC_PMCORR")
+        )
+    )
+
+    # Minimal columns to compute PM-corrected positions and do filtering.
+    if columns and COLUMNS_REQUIRED - set(columns):
+        raise ValueError(f"columns must include all of {COLUMNS_REQUIRED}")
```

Comment:
And this seems fine and is solid explicit coding style,  I'm just not sure if it would be a lighter touch to just add any needed columns to columns.